tooling/minimize/src/main.rs

```diff
@@ -120,10 +129,147 @@ macro_rules! show_error {
     ($($tt:tt)*) => { crate::show_error(&format_args!($($tt)*)) };
 }
 
+pub fn get_sysroot_dir() -> PathBuf {
+    match std::env::var_os("MINIRUST_SYSROOT") {
+        Some(dir) => PathBuf::from(dir),
+        None => {
+            let user_dirs = directories::ProjectDirs::from("org", "rust-lang", "minirust").unwrap();
+            user_dirs.cache_dir().to_owned()
+        }
+    }
+}
+
+fn be_rustc() {
+    // Get the rest of the command line arguments
+    let mut args: Vec<OsString> = env::args_os().skip(1).collect();
+
+    let use_panic_abort = {
+        let mut yes = false;
+        let mut it = args.iter();
+        while let Some(a) = it.next() {
+            if let Some(s) = a.to_str() {
+                if s == "--crate-name" {
+                    if let Some(next) = it.next().and_then(|o| o.to_str()) {
+                        if next == "panic_abort" {
+                            yes = true;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        yes
+    };
```

Comment:
This can be written more compactly using the `array_windows` iterator combinator. (It's unstable but that's okay, we're already using a bunch of unstable features here.)